backend/src/main/java/fk/prof/backend/aggregator/AggregationWindow.java

```diff
@@ -138,7 +134,7 @@ public void updateWorkInfoWithWSE(long workId, Recorder.Wse wse) {
     workInfo.updateWSESpecificDetails(wse);
   }
 
-  public void updateRecorderInfo(long workId, Recorder.RecorderInfo recorderInfo) {
+  public synchronized void updateRecorderInfo(long workId, Recorder.RecorderInfo recorderInfo) {
```

Comment:
Actually we can remove this synchronization, not really needed. workInfoLookup is not modifiable and here we are only updating a previously null value.